src/substitute.rs

```diff
@@ -75,10 +85,13 @@ pub fn prepare_request(
                         return Ok(ValueMissing {
                             key,
                             fallback: None,
+                            multiple: false,
                         });
                     };
 
-                    map.insert(key, value.clone());
+                    if let SubstitutionValue::Single(value) = value {
```

Comment:
I need to figure out of this

Reply:
I think this is okay for now, and then I will look into getting list support for graphql in a followup